File: Algorithms/Warmup/PlusMinus.java
package Algorithms.Warmup;
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;
/**
 * HackerRank Algorithms Warmup 6
 * https://www.hackerrank.com/challenges/plus-minus/problem
 * @author Hasol
 */
public class PlusMinus {
    // Complete the plusMinus function below.
    static void zeroPlusMinus(int[] arr) {
        double z = 0, p = 0, m = 0;
        for (int a: arr)
            if (a > 0)
                p++;
            else if (a < 0)
                m++;
            else
                z++;
        System.out.println (p / arr.length);
        System.out.println (m / arr.length);
        System.out.println (z / arr.length);
    }
    static final Scanner scanner = new Scanner(System.in);
    public static void main(String[] args) {
        int n = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");
        int[] arr = new int[n];
        String[] arrItems = scanner.nextLine().split(" ");
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");
        for (int i=0; i<n; i++)
            arr[i] = Integer.parseInt(arrItems[i]);
        zeroPlusMinus(arr);
        scanner.close();
    }
}

File: C/Introduction/plusminus.c
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

/*
 * HackerRank C Introduction 3
 * https://www.hackerrank.com/challenges/sum-numbers-c/problem
 * Author: Hasol
 */
int main ()
{
    int a, b;
    float c, d;
    scanf ("%d %d%*[\r\n]", &a, &b);
    scanf ("%f %f%*[\r\n]", &c, &d);
    printf ("%d %d\n", a+b, a-b);
    printf ("%.1f %.1f", c+d, c-d);
    return 0;
}
